src/dotnet/APIView/APIViewWeb/Repositories/ReviewManager.cs

```diff
@@ -207,48 +207,77 @@ public async Task<ReviewModel> GetReviewAsync(ClaimsPrincipal user, string id)
             return review;
         }
 
-        private async Task UpdateReviewAsync(ReviewModel review)
+        private async Task UpdateReviewOffline(ReviewModel review)
         {
+            var paramList = new List<ReviewGenPipelineParamModel>();
+            var languageService = GetLanguageService(review.Language);
             foreach (var revision in review.Revisions.Reverse())
             {
                 foreach (var file in revision.Files)
                 {
-                    if (!file.HasOriginal)
+                    //Don't include current revision if file is not required to be updated.
+                    // E.g. json token file is uploaded for a language, specific revision was already upgraded.
+                    if (!file.HasOriginal || file.FileName == null || !languageService.IsSupportedFile(file.FileName) || !languageService.CanUpdate(file.VersionString))
                     {
                         continue;
                     }
+                    paramList.Add(new ReviewGenPipelineParamModel()
+                    {
+                        FileID = file.ReviewFileId,
+                        ReviewID = review.ReviewId,
+                        RevisionID = revision.RevisionId,
+                        FileName = Path.GetFileName(file.FileName)
+                    });
+                }
+            }
+            await RunReviewGenPipeline(paramList, languageService.Name);
+        }
 
-                    try
+        private async Task UpdateReviewAsync(ReviewModel review)
+        {
+            var languageService = GetLanguageService(review.Language);
+            if (languageService == null)
+                return;
+
+            if (languageService.IsReviewGenByPipeline)
+            {
+                // Wait 30 seconds before running next review gen using pipeline to reduce queueing pipeline jobs
+                await Task.Delay(30000);
```

Comment:
Is this a predictably scheduled task, or something that gets queued up on demand? If the latter, tracking `UpdateReviewAsync` calls/delays via metrics would also be useful.

Reply:
If we cannot easily batch things then I'm OK with this for now but I'd like to move to a batch model here so this shouldn't be an issue in the future.

Reply:
It's only one time task when server restarted after a parser version upgrade

Reply:
I have filed an issue to make it batch model in a separate PR to unblock current reprocessing issue.